frontend/src/components/EntityList.vue

```diff
@@ -109,59 +61,56 @@ export default {
   },
   watch: {
     entitiesPerPage() {
-      this.getEntities({ resetPage: true });
+      this.getEntities({resetPage: true});
     },
+    schema() {
+      console.debug("Schema has changed?", this.schema)
+      if (this.schema === undefined || this.schema === null) {
+        console.debug("Oops, no valid schema")
+        return
+      }
+      this.getEntities();
+    }
   },
   methods: {
-    async selectSchema(schema) {
-      this.selectedSchema = schema;
-      this.filters = {};
-      const response = await api.getEntities({
-        schemaSlug: schema.slug,
-        limit: this.entitiesPerPage,
-        offset: this.offset,
-        meta: true,
-      });
-      const json = await response.json();
-      this.entities = json.entities;
-      this.totalEntities = json.total;
-      this.operators = json.meta.filter_fields.operators;
-      this.filterableFields = json.meta.filter_fields.fields;
-    },
     async changePage(page) {
       this.currentPage = page;
       await this.getEntities();
     },
-    async getEntities({ resetPage = false } = {}) {
+    async getEntities({resetPage = false} = {}) {
+      console.debug("Getting entities");
+      let _this = this;
       if (resetPage) {
         this.currentPage = 1;
       }
-      const response = await api.getEntities({
-        schemaSlug: this.selectedSchema.slug,
+      this.loading = true;
+      api.getEntities({
+        schemaSlug: this.schema.slug,
         limit: this.entitiesPerPage,
         offset: this.offset,
         filters: this.filters,
         orderBy: this.orderBy,
         ascending: this.ascending,
+      }).then(response => {
+        console.debug("Got entities", response)
+        _this.entities = response.entities;
+        _this.totalEntities = response.total;
+        _this.loading = false
       });
-      const json = await response.json();
-      this.entities = json.entities;
-      this.totalEntities = json.total;
+
     },
     async setFiltersAndSearch(filters) {
       this.filters = filters;
-      this.getEntities({ resetPage: true });
+      this.getEntities({resetPage: true});
     },
-    async reorder({ orderBy, ascending } = {}) {
+    async reorder({orderBy, ascending} = {}) {
       this.orderBy = orderBy;
       this.ascending = ascending;
       this.getEntities();
```

Comment:
Currently if you order by some field (not name) and switch to other schema, it be in constantly loading state. Quick research showed that placing 
```
this.orderBy = 'name';
this.ascending = true;
```
in `schema()` watch kind of helps but it looks like `EntityListTable` still keeps its state with initially ordered field

Reply:
Okay, I found it. In `updated()` of `EntityListTable` we are checking `this.schemaSlug`, but since you are now passing whole schema to component the check should be this instead:
```
updated() {
    if (this.previousSchema !== this.schema.slug) {
      this.previousSchema = this.schema.slug;
      this.ascending = true;
      this.orderBy = 'name';
    }
```

Reply:
Thanks for noticing. Fixed.